### ISSUE TITLE
 [Fix]: Best way to offer an MVP to consumers

### ISSUE THREAD

**Issue by omiras** [IronHack-2024/CodeQuestAPI#59]
@omiras After further investigation, I have seen that in fact there are a lot more than 5 “types” of question, including some with 8 possible answers, and even one “rogue” question that had 53 possible answers! LOL

To maintain coherence between the results in the responses given from both endpoints (https://codequestapi.onrender.com/api/v1/questions/ai, https://codequestapi.onrender.com/api/v1/questions/random), and after analysing the UI and UX when physically rendering 100s of questions in an app that consumes these endpoints, I strongly recommend that we only use two formats of question:

### Format 1:
questions with four possible answers and zero code examples:

<img width="574" alt="Screenshot 2024-12-10 at 04 54 30" src="https://github.com/user-attachments/assets/5d0be417-635b-4020-9767-46d5a4ec3c11">

### Format 2:
questions with four possible answers and one code example:

<img width="575" alt="Screenshot 2024-12-10 at 02 34 31" src="https://github.com/user-attachments/assets/55a81bd6-4d9b-4227-910b-f7200f19d068">

### How many Format 1 questions exist in our database?
The search below counted all the questions with 4 answers and 0 code examples in the `codequestapi-production` database in MongoDB and gave a result of 1217.
```
use('codequestapi-production');
db.getCollection('questions').countDocuments({
    "answerOptions": { $size: 4 },
    "codeExamples": { $size: 0 },
    "question": { $exists: true }
  });
```
### How many Format 2 questions exist in our database?
The search below counted all the questions with 4 answers and 1 code example in the `codequestapi-production` database in MongoDB and gave a result of 3449.
```
use('codequestapi-production');
db.getCollection('questions').countDocuments({
    "answerOptions": { $size: 4 },
    "codeExamples": { $size: 1 },
    "question": { $exists: true }
  });
```
### A solution for an MVP that can be offered to API consumers 
In conclusion, I believe the best way to offer an MVP to consumers is to simply delete (or exclude) all of the documents that do not correspond with the two formats mentioned previously from the `questions` collection in the `codequestapi-production` database in MongoDB. This solution does not address the `markdown2json` script issue, but it does mean that consumers can consume the API without further delay.

_Originally posted by @kevmhughes in https://github.com/IronHack-2024/CodeQuestAPI/issues/58#issuecomment-2530211427_

**Comment by omiras** [comment]
Hi @kevmhughes, 

This has been an extraordinary investigation on your part, identifying what's wrong and offering solutions. Thank you very much in advance.

That being said, I kindly ask you to address it, as it will bring significant value to the REST API.  
You can choose any of the proposed solutions. I am more inclined to delete the *wrong* imported programming questions because our production database shouldn't change for a while. The alternative of adding filters to **getRandomQuestionsDB** in **questions.services.js** to avoid malformed questions seems to me like [code smells](https://en.wikipedia.org/wiki/Code_smell).  

Please proceed with deleting the database documents. As a universal rule for database deletion, follow these steps:  

1. First, perform a query to retrieve the desired documents to be removed.  
2. Then, use the same query criteria to delete those documents.

**Comment by kevmhughes** [comment]
Hello @omiras, 

It seems that somebody has already modified the `codequestapi-production` database in MongoDB. Nevertheless, some documents still existed that did not correspond with the two formats mentioned previously. I have deleted them from the database.

Here you have screenshots demonstrating the actions taken:

1) Count total of documents type "format 1":
<img width="1188" alt="Screenshot 2024-12-10 at 14 52 13" src="https://github.com/user-attachments/assets/bbeda850-a46c-4943-9b8f-87709cbb9ce9">

2) Count total of documents type "format 2":
<img width="1188" alt="Screenshot 2024-12-10 at 14 52 26" src="https://github.com/user-attachments/assets/40502805-21ad-4b17-9360-f6612e39922a">

3) Count total of all documents:
<img width="1188" alt="Screenshot 2024-12-10 at 14 53 15" src="https://github.com/user-attachments/assets/0745feb5-4ccb-4ec8-a1fe-7f4c9545d070">

4) Count total of documents that do not correspond with "format 1" and "format2" questions:
<img width="1188" alt="Screenshot 2024-12-10 at 14 53 53" src="https://github.com/user-attachments/assets/a861f52f-37a4-427f-9bd6-4c1916bc44eb">

5) Delete all documents that do not correspond with "format 1" and "format2:
<img width="1188" alt="Screenshot 2024-12-10 at 14 55 27" src="https://github.com/user-attachments/assets/700640c8-ef8b-47f5-ab53-84e2d4d8bd77">

6) Count total documents after deletion to ensure only the unwanted documents were deleted:
<img width="1188" alt="Screenshot 2024-12-10 at 14 58 13" src="https://github.com/user-attachments/assets/9c03d3c6-9e62-4f6a-84e1-9e95d364b7bf">

**Comment by kevmhughes** [comment]
I am pleased to confirm that after removing the" smelly" code from the `codequestapi-production` database in MongoDB our API consumers will have a cleaner and highly improved UI and UX! 🥳🥳🥳

### The two formats rendered from: https://codequestapi.onrender.com/api/v1/questions/ai:

Format 1:
<img width="765" alt="Screenshot 2024-12-10 at 15 31 09" src="https://github.com/user-attachments/assets/8fa9aaae-b1d5-4fb9-ab39-6945325d5b0a">

Format 2: 
<img width="765" alt="Screenshot 2024-12-10 at 15 30 56" src="https://github.com/user-attachments/assets/e87a0127-1b86-4d9c-bc92-13624f351136">

### The two formats rendered from: https://codequestapi.onrender.com/api/v1/questions/random:

Format 1: 
<img width="765" alt="Screenshot 2024-12-10 at 15 30 11" src="https://github.com/user-attachments/assets/824ba4ca-d6f7-42d7-9b89-fc36f6114007">

Format 2:
<img width="765" alt="Screenshot 2024-12-10 at 15 29 31" src="https://github.com/user-attachments/assets/321308e1-2fbf-4094-a86d-1d3a958ce142">

**Comment by omiras** [comment]
That's amazing! Hurray! We are one step closer to fully delivering this to the world.
